packages/@core.keystone/fields/AutoIncrementInteger/Implementation.js

```diff
@@ -31,10 +35,32 @@ class AutoIncrementIntegerKnexFieldAdapter extends Integer.adapters.knex {
         const tableName = this.listAdapter.tableName
         const fieldName = this.dbPath
         const knex = this.listAdapter.parentAdapter.knex
-        let [{ max }] = await knex(tableName).max(fieldName)
-        max = max || 0
-        return ++max
-    }    
+        // NOTE(pahaz): it's really hack! please don't copy this staff in a future! I'll find a better solution for that!
+        //  I didn't found a way to use knex subquery for that. Probably, we need to create some DB procedure or use another table with
+        //  sequence column. At the moment this code just help us to avoid `duplicate key value violates unique constraint`
+        const rlock = new Redlock([taskQueue.client])
+        const redisLockKey = `AutoIncrementInteger:${tableName}:${fieldName}`
+        const redisMaxValueKey = `AutoIncrementInteger:${tableName}:${fieldName}:value`
+        let lock = await rlock.acquire([redisLockKey], 500) // 0.5 sec
+        try {
+            let currentMaxNumber = await taskQueue.client.get(redisMaxValueKey)
+            if (!currentMaxNumber) {
+                let [{ max }] = await knex(tableName).max(fieldName)
+                currentMaxNumber = max
+            } else {
+                // NOTE(pahaz): if someone set new number value by hands we need to generate next number MAX + 1
+                let [{ max }] = await knex(tableName).max(fieldName)
+                if (max > currentMaxNumber) {
+                    currentMaxNumber = max
+                }
+            }
+            currentMaxNumber = parseInt(currentMaxNumber || 0)
+            await taskQueue.client.set(redisMaxValueKey, currentMaxNumber + 1)
+            return currentMaxNumber + 1
+        } finally {
+            await lock.release()
+        }
+    }
```

Comment:
Why not to use sequences in database?

Reply:
Because you can set the number by hands. And we probably will change this logic in a future. We want to use organization specific numbers. Uniq for organization

Reply:
I don't found a way to use it without too match hacks ...